docker-images/ubuntu/Dockerfile

```diff
@@ -61,11 +62,22 @@ ADD codebuild codebuild
 RUN set -eux; \
   export LD_LIBRARY_PATH=""; \
   . codebuild/bin/s2n_setup_env.sh; \
-  codebuild/bin/s2n_install_test_dependencies.sh; \
+  export PATH=$TEST_DEPS_DIR/clang/bin:$PATH; \
+  TESTS=integration codebuild/bin/s2n_install_test_dependencies.sh; \
+  TESTS=benchmark codebuild/bin/s2n_install_test_dependencies.sh; \
+  TESTS=fuzz codebuild/bin/s2n_install_test_dependencies.sh; \
+  TESTS=unit BUILD_S2N=false S2N_LIBCRYPTO=openssl-1.0.2 codebuild/bin/install_default_dependencies.sh; \
+  TESTS=unit BUILD_S2N=false S2N_LIBCRYPTO=openssl-1.0.2-fips codebuild/bin/install_default_dependencies.sh; \
+  TESTS=unit BUILD_S2N=false S2N_LIBCRYPTO=libressl codebuild/bin/install_default_dependencies.sh; \
+  TESTS=unit BUILD_S2N=false S2N_LIBCRYPTO=boringssl codebuild/bin/install_default_dependencies.sh; \
   rm -rf /var/lib/apt/lists/*; \
   apt-get clean; \
+  rm -rf /tmp/*; \
   echo done
 
+ARG OPENSSL_VERSION=openssl-1.1.1
+ENV S2N_LIBCRYPTO=${OPENSSL_VERSION}
```

Comment:
Rather than taking up a bunch of space on your local machine, this splits the layers for the different openssl versions very last so it all has the same base image with only an environment variable changed. This does increase the size of the image, but it ends up working out better than having a full layer for each of the versions.